graylog2-server/src/main/java/org/graylog/plugins/views/search/rest/FieldTypesResource.java

```diff
@@ -72,4 +76,15 @@ public Set<MappedFieldTypeDTO> byStreams(@ApiParam(name = "JSON body", required
         final ImmutableSet<String> streams = searchUser.streams().readableOrAllIfEmpty(request.streams());
         return mappedFieldTypesService.fieldTypesByStreamIds(streams, request.timerange().orElse(RelativeRange.allTime()));
     }
+
+    @POST
+    @ApiOperation(value = "Trigger a full refresh of field types")
+    @Path("/poll")
+    @RequiresPermissions("*")
+    @AuditEvent(type = FIELD_TYPE_POLLING_TRIGGERED)
+    public Response triggerFieldTypePolling() {
+        fieldTypePoller.triggerFullRefresh();
```

Comment:
Please correct me if I am wrong, but having manual trigger and periodical triggers in the same time, and `lastFullRefresh` in periodical updated at the end of `doRun` invocation, we probably risk unnecessary invocations of full refresh (i.e. periodical call is started during execution of manually triggered call). 
I assume we just accept that, don't want to over-complicate  and prevent that, as manual calls are supposed to be used rarely/exceptionally?

Reply:
Yes, right now they should be used only in tests or when debugging. Also, we do have a [check](https://github.com/Graylog2/graylog2-server/blob/master/graylog2-server/src/main/java/org/graylog2/indexer/fieldtypes/IndexFieldTypePollerPeriodical.java#L187) in place which skips subsequent polls if there is one in progress right now.